### PR TITLE
test: validate package

### PR DESCRIPTION
**Pull request by AustinAbro321**
## Description

This PR adds tests to validate package

## Checklist before merging

- [ ] Test, docs, adr added or updated as needed
- [ ] [Contributor Guide Steps](https://github.com/defenseunicorns/zarf/blob/main/.github/CONTRIBUTING.md#developer-workflow) followed

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,6 +1,6 @@
 module github.com/defenseunicorns/zarf
 
-go 1.21.8
```
Comment (phillebaba) [inline go.mod:3]: Bump Go version in a separate PR.

File: src/types/validate_test.go [hunk 2]
```diff
@@ -0,0 +1,494 @@
...
+	"github.com/defenseunicorns/zarf/src/config/lang"
+	"github.com/defenseunicorns/zarf/src/pkg/variables"
+	"github.com/stretchr/testify/require"
+)
+
+func TestZarfPackageValidate(t *testing.T) {
+	tests := []struct {
+		name     string
+		pkg      ZarfPackage
+		wantErrs []string
```
Comment (phillebaba) [inline src/types/validate_test.go:23]: Rename this to expectedErrs

**Review by phillebaba — CHANGES_REQUESTED** [review]

File: src/types/validate_test.go
Comment (lucasrod16) [inline src/types/validate_test.go:1]: We should be able to run all of these test functions and test cases in parallel with `t.Parallel()`

**Review by lucasrod16 — CHANGES_REQUESTED** [review]

File: src/types/validate_test.go
Comment (lucasrod16) [inline src/types/validate_test.go:1]: Could you also ensure that the top-level test functions are run in parallel?

**Review by lucasrod16 — CHANGES_REQUESTED** [review]

**Review by lucasrod16 — DISMISSED** [review]
lgtm

File: src/types/validate.go [hunk 3]
```diff
@@ -40,27 +40,28 @@ func SupportedOS() []string {
 
 // Validate runs all validation checks on the package.
 func (pkg ZarfPackage) Validate() error {
+	errs := []error{}
 	if pkg.Kind == ZarfInitConfig && pkg.Metadata.YOLO {
-		return fmt.Errorf(lang.PkgValidateErrInitNoYOLO)
+		errs = append(errs, fmt.Errorf(lang.PkgValidateErrInitNoYOLO))
```
Comment (schristoff-du) [inline src/types/validate.go:45]: Could we use `errors.Join` as an append instead of making a `errs := []error{}`
I would say either do a `var err error` and then `errors.Join(error, fmtErrorf(...)`

File: zarf.schema.json [hunk 4]
```diff
@@ -1003,7 +1003,7 @@
       "properties": {
         "name": {
           "type": "string",
-          "pattern": "^[a-z0-9\\-]*[a-z0-9]$",
+          "pattern": "^[a-z0-9][a-z0-9\\-]*$",
```
Comment (schristoff-du) [inline zarf.schema.json:1006]: What drove this change?

File: src/types/validate_test.go [hunk 5]
```diff
@@ -0,0 +1,506 @@
...
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		tt := tt
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			err := tt.component.Validate()
+			if tt.expectedErrs == nil {
```
Comment (schristoff-du) [inline src/types/validate_test.go:498]: I nitted Wayne about it may be better to make expectedErrs a bool, but it's up to you.

**Review by schristoff-du — COMMENTED** [review]
Couple quick questoins

File: src/types/validate_test.go (hunk 5 above)
Comment (phillebaba) [reply src/types/validate_test.go:498]: I think this is better to avoid situations where expectedErrs is not empty but the bool is false.

File: zarf.schema.json (hunk 4 above)
Comment (AustinAbro321) [reply zarf.schema.json:1006]: It's the wrong regex, we currently do a check here https://github.com/defenseunicorns/zarf/blob/main/src/types/validate.go#L26 at create time. I'm going to refactor to use the schema instead of a separate test, but wanted to get the schema in line and create tests first

Comment (schristoff) [reply zarf.schema.json:1006]: Dope :)

**Review by schristoff-du — APPROVED** [review]

**Review by lucasrod16 — APPROVED** [review]

**Review by phillebaba — APPROVED** [review]
